### PR TITLE
Closes #52 HWSD2

### LINKED ISSUE #52: New/Replacement Harmonized World Soils Database version 2.0
A new version of HWSD v.2 was released in 2023, with a minor update to 2.01 in September 2023. It is described at https://gaez.fao.org/pages/hwsd

The report for the data is available at https://www.fao.org/3/cc3823en/cc3823en.pdf

image

### PR DESCRIPTION
**Pull request by msteckle**
This pull request would close issue #52 for the harmonized world soils database version 2. The script produces a netcdf at 0.5 degree resolution for cSoil representing 0--200cm soil depth. 

**Important notes:**
- cSoil: carbon stock in kg m-2 for all soil pools (weighted mean) and all soil levels (0--200cm)
- pool: in HWSDv2, I think this would be the the different soil types in a region (e.g., mollisol), whose fractional dominance is indicated via the SEQUENCE variable in the HWSDv2 database.
- level: the depth of a soil layer (e.g., D1 = 0--20cm)
- cSoilLevels: carbon stock in kg m-2 for all soil pools (weighted mean) and a singular soil level

**Questions**

- Each model might have a different definition of cSoil, e.g., one may have a soil profile down to 500cm while another only goes to 30cm. Shouldn't ILAMB be able to aggregate the benchmark to the best matching cSoil of the model?
    - In other words, should I get cSoil for each level? 
        - Could be one netcdf with multiple variables like cSoilLevels0to20, cSoilLevels20to40, etc. or many individual netcdfs.

**Result**

download

### CODE REVIEW

**Comment by nocollier** [comment]
I think we merge this as is. Your questions are good but as I understand the situation part of the ambiguity in benchmarking soil carbon. Maybe we need to look at cSoil in CMIP6 models, but my understanding is that it isn't report stratified in any model. There are only some models who also report cSoilAbove1m. Very nicely done! You code beautifully!

**Comment by msteckle** [comment]
> I think we merge this as is. Your questions are good but as I understand the situation part of the ambiguity in benchmarking soil carbon. Maybe we need to look at cSoil in CMIP6 models, but my understanding is that it isn't report stratified in any model. There are only some models who also report cSoilAbove1m. Very nicely done! You code beautifully!

Thanks, sounds good to me!
